src/triage/component/catwalk/storage.py

```diff
@@ -213,123 +243,179 @@ def matrix_with_sorted_columns(self, columns):
                     Columnset and desired columnset mismatch. Unique items: %s
                 ''', columnset ^ desired_columnset)
 
-    def save_yaml(self, df, project_path, name):
-        with smart_open.smart_open(os.path.join(project_path, name + ".yaml"), "wb") as f:
-            yaml.dump(df, f, encoding='utf-8')
+    def save_metadata(self, df, project_path, name):
+        path_parsed = urlparse(project_path)
+        scheme = path_parsed.scheme  # If '' of 'file' is a regular file or 's3'
+        if not scheme or scheme == 'file':  # Local file
+            with open(os.path.join(project_path, name + ".yaml"), "wb") as f:
+                yaml.dump(df, f, encoding='utf-8')
+        elif scheme == 's3':
```

Comment:
All of these if statements are definitely crying out for some refactoring, but we can do that in a future commit.

The direction I'm thinking is that S3Store, FSStore are too specific to model pickles and be repurposed into something generic enough to use here. Then `MatrixStore` takes a storage class, and there we go.

Also, the list of supported schemes is probably in too many places. A class-level `SUPPORTED_SCHEMES` should work as not too invasive of a change.

Reply:
That's a good idea, for the matrix store to simply take a storage backend.